wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -219,11 +216,13 @@ public FactoryDto getFactory(
       }
     }
 
-    final List<FactoryDto> factories = new ArrayList<>();
-    for (Factory factory : factoryManager.getByAttribute(maxItems, skipCount, query)) {
-      factories.add(injectLinks(asDto(factory)));
+    Page<? extends Factory> factoriesPage =
+        factoryManager.getByAttribute(maxItems, skipCount, query);
+    List<FactoryDto> list = new ArrayList<>();
+    for (Factory factory : factoriesPage.getItems()) {
+      list.add(injectLinks(asDto(factory)));
     }
-    return factories;
+    return Response.ok().entity(list).header("Link", createLinkHeader(factoriesPage)).build();
```

Comment:
```
    return Response.ok()
        .entity(
            workspacesPage
                .getItems()
                .stream()
                .filter(ws -> status == null || status.equalsIgnoreCase(ws.getStatus().toString()))
                .map(DtoConverter::asDto)
                .collect(toList()))
        .header("Link", createLinkHeader(workspacesPage))
        .build();
```

Reply:
Not possible due to checked exceptions thrown in FactoryService::asDto